Remove `react-datetime` and `moment` packages

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/5010

The only place I saw that the older DateTime picker is still used is the member settings page. I've changed that to use the native date input. 

![datepicker](https://user-images.githubusercontent.com/12435965/144674198-f714879a-4de1-49b4-9e22-c17cc3ae8fba.gif)

Betree reviewed (commented):
This field is mandatory, can we enforce that in the interface?

![image](https://user-images.githubusercontent.com/1556356/144802190-3235a2e5-4b1c-4c62-ae9f-505d8466a2eb.png)

SudharakaP commented on components/InputField.js:
```diff
@@ -319,57 +308,6 @@ class InputField extends React.Component {
-                <Box width={[1, 10 / 12]}>
-                  <DateTimePicker
-                    name={field.name}
-                    timeFormat={field.timeFormat || timeFormat}
```
Thanks. Just removed this one. 👍🏽

SudharakaP commented:
> This field is mandatory, can we enforce that in the interface?
> 
> ![image](https://user-images.githubusercontent.com/1556356/144802190-3235a2e5-4b1c-4c62-ae9f-505d8466a2eb.png)

This one is only there on Firefox (Chrome doesn't seem to have this clear button by default) and I tried to remove this last week but unfortunately it doesn't seem obvious and somehow I couldn't figure it out. Let me try again. 🤔

Betree commented:
> > This field is mandatory, can we enforce that in the interface?
> > ![image](https://user-images.githubusercontent.com/1556356/144802190-3235a2e5-4b1c-4c62-ae9f-505d8466a2eb.png)
> 
> This one is only there on Firefox (Chrome doesn't seem to have this clear button by default) and I tried to remove this last week but unfortunately it doesn't seem obvious and somehow I couldn't figure it out. Let me try again. thinking

I just tried on https://developer.mozilla.org/fr/docs/Web/HTML/Element/Input/datetime-local, setting the `required` attribute removes the ability to clear the field.

SudharakaP commented:
> > > This field is mandatory, can we enforce that in the interface?
> > > ![image](https://user-images.githubusercontent.com/1556356/144802190-3235a2e5-4b1c-4c62-ae9f-505d8466a2eb.png)
> > 
> > 
> > This one is only there on Firefox (Chrome doesn't seem to have this clear button by default) and I tried to remove this last week but unfortunately it doesn't seem obvious and somehow I couldn't figure it out. Let me try again. thinking
> 
> I just tried on https://developer.mozilla.org/fr/docs/Web/HTML/Element/Input/datetime-local, setting the `required` attribute removes the ability to clear the field.

@Betree : Are you sure? I tried this but on Firefox (at least in my version it doesn't work) and also with the required attribute the input box turns green when one clears it out; which I didn't like. 🤔 

![image](https://user-images.githubusercontent.com/12435965/145042731-2612c357-dbcc-4bea-af2c-653291e3a548.png)

Betree commented:
@SudharakaP https://www.loom.com/share/6d0320663f2749afa527ad178da23186

Betree commented:
@SudharakaP Where is this green style coming from? Can you inspect the styles and remove the root cause?

SudharakaP commented:
> @SudharakaP Where is this green style coming from? Can you inspect the styles and remove the root cause?

The green color is coming from the `InputField` actually so I fixed that one and also added the required attribute. But still the main reason I didn't do this was in my Firefox I don't see this clear button dissappearing.  Out of curiosity what version of Firefox you use? 🤔

Betree commented:
> > @SudharakaP Where is this green style coming from? Can you inspect the styles and remove the root cause?
> 
> The green color is coming from the `InputField` actually so I fixed that one and also added the required attribute. But still the main reason I didn't do this was in my Firefox I don't see this clear button dissappearing. Out of curiosity what version of Firefox you use? thinking

Firefox 94.0. Do you see it on https://codesandbox.io/s/funny-river-g362l?file=/src/App.js?

SudharakaP commented:
> > > @SudharakaP Where is this green style coming from? Can you inspect the styles and remove the root cause?
> > 
> > 
> > The green color is coming from the `InputField` actually so I fixed that one and also added the required attribute. But still the main reason I didn't do this was in my Firefox I don't see this clear button dissappearing. Out of curiosity what version of Firefox you use? thinking
> 
> Firefox 94.0. Do you see it on https://codesandbox.io/s/funny-river-g362l?file=/src/App.js?

Sorry I realized this is my mistake. 😢  We need to pass the required field to the `FieldGroup` component which I somehow missed. Now it's corrected and the clear button is gone.

Betree commented on components/InputField.js:
```diff
@@ -181,10 +170,12 @@ class InputField extends React.Component {
-    if (!value) {
+    const type = this.props.type || 'text';
+
+    if (!value && type !== 'datetime-local') {
```
Why ignore the `this.props.required` flag for `datetime-local`?

SudharakaP replied:
Ops, sorry I was trying something and committed this change by mistake. This should not have been changed. I've reverted this. 🤔 🤯

Betree commented:
I'm still able to clear the input for the member modal. The form doesn't warn me and it produces a strange result

![image](https://user-images.githubusercontent.com/1556356/145370122-bbe5081c-bd37-428c-bc03-db651ea1339a.png)

SudharakaP commented:
> I'm still able to clear the input for the member modal. The form doesn't warn me and it produces a strange result

Ops, I forgot somehow about this member modal. I've corrected this and added the required attribute there as well. Thanks.

Betree commented:
I still end up with strange styles & behaviors on this page, both Chrome and FF (tho different issues)

https://user-images.githubusercontent.com/1556356/145576753-0534117e-3bc9-4694-a866-1bccc995eca1.mp4


https://user-images.githubusercontent.com/1556356/145576810-c5ab4dd3-e81e-450b-a38c-d934c10c635e.mp4

SudharakaP commented:
> I still end up with strange styles & behaviors on this page, both Chrome and FF (tho different issues)

Sorry, there were some inconsistencies with Chrome with the scroll bar appearing. I've fixed that as well as disabled backspace as we are not supposed to clear out the form.

Betree commented:
Can we improve this error message?

![image](https://user-images.githubusercontent.com/1556356/145798662-610853af-b55b-4908-9b03-a976cdd091af.png)

SudharakaP commented:
> Can we improve this error message?

@Betree : I've done some improvements both to this members modal as well as the edit event pages so that we show the error message when the user inserts and invalid date. This is only possible if they manually input a very large date or a something like 0000-00-00. Anyways, I think,

1) We could ideally do some refactoring on validating the collective settings section before the user clicks on save; for example if the collective name is empty etc. 

2) Maybe I was thinking it would be nice to disable the save button in `MemberForm` if the date is invalid. But I didn't find a simpler way to do this without refactoring the whole section. It seems that the submit button lies in `EditMemberModal`. If you have some ideas on this let me know and maybe we can tackle this on a new issue? 🤔

Betree reviewed (commented):
> We could ideally do some refactoring on validating the collective settings section before the user clicks on save; for example if the collective name is empty etc.

That would be ideal, and show an error directly on the field.

> Maybe I was thinking it would be nice to disable the save button in MemberForm if the date is invalid. But I didn't find a simpler way to do this without refactoring the whole section. It seems that the submit button lies in EditMemberModal. If you have some ideas on this let me know and maybe we can tackle this on a new issue? thinking

We try to avoid this pattern, as it often leads up to having the "Submit" button disabled and the user doesn't know why. See https://ux.stackexchange.com/a/76306 for more context.

In the demo you shared, the error is clearly surfaced so I'm ok with the implementation. It's a great improvement on what we have.

Just a minor feedback below about using best practices for Formik validations and I think we'll be good to merge!

SudharakaP commented on components/edit-collective/sections/MemberForm.js:
```diff
@@ -113,14 +128,33 @@ const MemberForm = props => {
-                      setFieldValue(field.name, value);
+                      if (field.name === 'since') {
+                        if (dayjs(value).isValid()) {
+                          setValidDate(true);
```
@Betree : Thanks much for the review and the feedback. 🤗  I've refactored the code to use Formik validation on the MemberForm. 

I've opened a new issue for the other parts of the form validation which I see could use some refactoring; https://github.com/opencollective/opencollective/issues/5046

Let me know if this is good to merge now? 😃

Betree reviewed (approved):
I haven't re-tested, but the code looks good to me. Ok to merge if both forms are working properly!
